javascript/javascript1/week2/candy_helper.js

```diff
@@ -0,0 +1,52 @@
+const candies = {
+  Sweet: 0.5,
+  Chocolate: 0.7,
+  Toffee: 1.1,
+  "Chewing-gum": 0.03,
+};
+
+const boughtCandyPrices = [];
+let amountToSpend = +(Math.random() * 100).toFixed(2); // + converts str back to the number
+
+function addCandy(candyType, weight) {
+  const price = +(candies[candyType] * weight).toFixed(2); // + converts str back to the number
+  let budget = amountToSpend - price;
+  if (budget > 0 && canBuyMoreCandy()) {
+    boughtCandyPrices.push(price);
+    amountToSpend = budget;
+    return `You can buy more, so please do! You have ${amountToSpend.toFixed(
+      2
+    )} to spend.`;
+  } else {
+    return `You are buying candy that costs ${price}.You have only ${amountToSpend.toFixed(
+      2
+    )} left. Enough candy for you?!`;
+  }
+}
```

Comment:
Good job!

This is a limited and self containers example, so it does not matter too much, but try in the future to not have your function return the log messages. try to make your functions do the "logic" and then have the logging outside of the function. Along the same lines, all the `toFixed` is for having "nice" output logs at the end. Could you live with only a `toFixed` at the very end after the logic and outside of the function? You can google "Separation of concerns", which is a concept that one wants to separate for example logic from rendering/outputting.

Reply:
Yes! That makes sense!